libphobos/m4/druntime/libraries.m4

```diff
@@ -16,6 +40,17 @@ AC_ARG_ENABLE(thread-lib,
 ])
 
 
+# DRUNTIME_LIBRARIES_TIME
+# -----------------------
++# Check whether -lrt is required for sched_yield, nanosleep, or clock_gettime.
+AC_DEFUN([DRUNTIME_LIBRARIES_TIME],
+[
+AC_SEARCH_LIBS([sched_yield], [rt])
+AC_SEARCH_LIBS([nanosleep], [rt])
+AC_SEARCH_LIBS([clock_gettime], [rt])
+])
+
```

Comment:
Do we actually need this? If `librt` is required, the driver should still link it automatically as GCC will define `TIME_LIBRARY`. We could probably rewrite this to use gdc instead and simply assert that the functions are available?

The main problem with this approach is that it won't work for static libraries. It works for `librt` as we have hardcoded support for that in the driver code, but `ws2_32` for example will not get linked into user programs automatically. Ironically all our tests should pass (libtool makes sure to encode dependencies into the `.la` file which even works for static libraries) but user programs wouldn't link `ws2_32` (as gdc directly adds `-lphobos` and therefore does not use the libtool `.la` file). 

Turns out this is a problem for libdl already: As the static libphobos also uses `dlopen` for curl there's a link error. 

```d
import std.net.curl, std.stdio;
void main()
{
    writeln(get("google.com"));
}
// /opt/gdc/bin/gdc test.d -shared-libphobos => ok
// /opt/gdc/bin/gdc test.d => link error
```
As one possible solution: Could we dump the required libraries to a file and parse that file in the driver?